src/main/scala/firrtl/passes/Checks.scala

```diff
@@ -57,6 +57,11 @@ object CheckHighForm extends Pass {
     s"$info: [module $mname] Primop $op lsb $lsb > $msb.")
   class NonLiteralAsyncResetValueException(info: Info, mname: String, reg: String, init: String) extends PassException(
     s"$info: [module $mname] AsyncReset Reg '$reg' reset to non-literal '$init'")
+  class ResetInputException(info: Info, mname: String, expr: Expression) extends PassException(
+    s"$info: [module $mname] Abstract Reset not allowed as top-level input: ${expr.serialize}")
+  class ResetExtModuleOutputException(info: Info, mname: String, expr: Expression) extends PassException(
+    s"$info [module $mname] Abstract Reset not allowed as ExtModule output: ${expr.serialize}")
```

Comment:
Is it ok to have an input ExtModule reset type? I think it is, but I think its worth discussing?

Reply:
I'm actually not sure, is it okay? @edwardcwang has requested the ability to have the reset values for `AsyncReset` registers be inputs from the top or ExtModules since it might be coming from some analog (or otherwise not expressible in FIRRTL) module, I guess this could be a related feature.

Reply:
I mean, is there a particularly good reason to restrict it?